lib/connection.js

```diff
@@ -1137,28 +1151,112 @@ Connection.prototype.search = function(sosl, callback) {
  *
  * @typedef {Object} DescribeSObjectResult
  */
+/**
+ * Parameter for describeSObject call
+ * 
+ * @typedef {Object} DescribeSObjectOptions
+ */
 /**
  * Synonym of Connection#describe()
  *
  * @method Connection#describeSObject
- * @param {String} type - SObject Type
+ * @param {String|DescribeSObjectOptions} type - SObject Type
  * @param {Callback.<DescribeSObjectResult>} [callback] - Callback function
  * @returns {Promise.<DescribeSObjectResult>}
  */
 /**
  * Describe SObject metadata
  *
  * @method Connection#describe
- * @param {String} type - SObject Type
+ * @param {String|DescribeSObjectOptions} type - SObject Type
  * @param {Callback.<DescribeSObjectResult>} [callback] - Callback function
  * @returns {Promise.<DescribeSObjectResult>}
  */
 Connection.prototype.describe =
 Connection.prototype.describeSObject = function(type, callback) {
-  var url = [ this._baseUrl(), "sobjects", type, "describe" ].join('/');
-  return this.request(url).thenCall(callback);
+  var name = type.type ? type.type : type;
+  var url = [ this._baseUrl(), "sobjects", name, "describe" ].join('/');
+  var headers = type.ifModifiedSince 
+    ? { 'If-Modified-Since': type.ifModifiedSince } 
+    : {};
+  return this.request({
+    method: 'GET',
+    url,
+    headers
+  }).then(resp => {
+    if (resp === '') {
+      return Promise.resolve(undefined);
+    } else {
+      return Promise.resolve(resp);
+    }
+  }).thenCall(callback);
 };
 
+/**
+ * Result returned by batchDescribeSObjects call
+ *
+ * @typedef {Object[]} DescribeSObjectResult
+ */
+/**
+ * Synonym of Connection#batchDescribe()
+ *
+ * @method Connection#batchDescribeSObjects
+ * @param {String[]} types - SObject Type
+ * @param {number} next - the next index in the types array to process (max batch size is 25)
```

Comment:
So the consumer has to set the next each time they call this if `types` is more than 25?

Reply:
Yes. I was back and forth about this, and I'm still on the fence. This method could handle the multiple batch requests itself, but it would make it harder to debug when a problem occurs.

Reply:
You could maybe have it be an option, like autofetch is? https://github.com/jsforce/jsforce/blob/master/lib/query.js#L29

Reply:
Ok I tried it that way. See the `feat(conn): autofetch for batchDescribe` commit.